### PR TITLE
[VYMCA-79] New CSV auth for user segmentation

### PR DESCRIPTION
**Pull request by hamrant**
### Steps for review (Add filters within the shared content server):
Jira - https://fivejars.atlassian.net/browse/VYMCA-73
Basecamp - https://3.basecamp.com/3608022/buckets/17045978/todos/3035561517

- [x] Login as admin
- [x] Go to `/admin/modules`
- [x] Enable `Virtual Y Shared Content` module
- [x] Go to `/admin/virtual-y/shared-content/server/openy`
- [x] Press save to generate token
- [x] Go to fetch page `/admin/virtual-y/shared-content/server/openy/fetch`
- [x] Check that you can see filters and they works fine

Screenshot from 2020-09-23 13-59-46

**Note:** There no sorting filter by `YMCAS using content`, we need to add a category column, and for MVP `Donated By` is a text field instead of a select list

### Steps for review (New CSV auth for user segmentation):
Jira - https://fivejars.atlassian.net/browse/VYMCA-79
Basecamp - https://3.basecamp.com/3608022/buckets/17045978/todos/3047898482

- [x] Login as admin
- [x] Go to `/admin/openy/virtual-ymca/gc-auth-settings`
- [x] Enable **`Custom provider`** and disable `Dummy provider`
- [x] Go to `/admin/config/people/captcha/recaptcha` and add site keys (configured for domain **ci.fivejars.com**):

Site key:
```
6LfCiMoZAAAAAKTLI7VrO5Lx4HD7KowBTytsBorJ
```
Secret key:
```
6LfCiMoZAAAAAAd9aEhHeEod-4YhO7qfyDOZfD0P
```

If something wrong with ReCaptcha - add a site here: https://www.google.com/recaptcha/admin and use your keys instead

- [x] Go to `/admin/people`
- [x] Check that you can see here demo users with `Virtual YMCA` role and all of them have the status `Blocked`
- [x] Adit any user and copy email
- [x] Open `/virtual-y-login` page in incognito or different browser
- [x] Check that you can see the login form with an email field and ReCaptcha (styles will be fixed later)
- [x] Try to use invalid or wrong email, check that validation works fine and you can see an error
- [x] Try to login with correct email (for example `johndoe@test.com`)
- [x] Check that you can see the message about the verification link

Screenshot from 2020-09-23 14-19-25

- [x] Check that back to login form link works fine
- [x] Go to http://mail.pr-109.openy-gated-content.sandbox-2.ci.fivejars.com
- [x] Check that you can see a message with a login link

Screenshot from 2020-09-23 14-20-40

- [x] Open link in incognito
- [x] Check that you logged-in and redirected to `/virtual-ymca#/`
- [x] As admin go to `/admin/openy/virtual-ymca/gc-auth-settings/provider/custom`
- [x] Uncheck `Enable ReCaptcha` and `Enable Email verification`
- [x] Try to login with different user and check that on login form no ReCaptcha and there no verify email step

### CODE REVIEW

**Review by anpolimus — APPROVED** [review]

**Review by NightHunterSV — APPROVED** [review]
Code looks fine

File: openy_gated_content.post_update.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      $gated_content_login_page = \Drupal::entityTypeManager()
+        ->getStorage('node')
+        ->create([
+          'type' => 'landing_page',
+          'title' => 'VIRTUAL Y Login',
+          'field_content' => [$gated_content_login_prgf],
+        ]);
+      $gated_content_login_page->save();
+    }
+    $gated_content_login_alias = $path_alias_manager->getAliasByPath('/node/' . $gated_content_login_page->id());
```
Comment (NightHunterSV) [inline openy_gated_content.post_update.php:55]: We can reduce the code duplication, moving this piece of code out of the conditional section.

Comment (hamrant) [reply openy_gated_content.post_update.php:55]: no

Comment (hamrant) [reply openy_gated_content.post_update.php:55]: ok, let's do this

Comment (hamrant) [reply openy_gated_content.post_update.php:55]: Fixed

**Review by NightHunterSV — APPROVED** [review]

**Review by NightHunterSV — APPROVED** [review]

**Review by anpolimus — APPROVED** [review]

**Comment by hamrant** [comment]
retest

**Comment by hamrant** [comment]
retest

**Pull request merged**